javascript/javascript1/week1/app.js

```diff
@@ -0,0 +1,73 @@
+//STEP 2
+//user freecodecamp fccdc38edef-ba87-461c-96cd-bf07dcfc8098
+
+//STEP 3
+
+/*Age-ify*/
+var yearOfBirth = 1987; 
+var yearFuture =  2027;
+var age = yearFuture - yearOfBirth;
+console.log("You will be " + age + " years old in " + yearFuture + ".");
+
+/*Goodboy-Oldboy */
+var dogYearOfBirth = 2020;
+var dogYearFuture = 2030;
+var dogYear = dogYearFuture - dogYearOfBirth;
+var shouldShowResultInDogYearsld = true;
+
+if(shouldShowResultInDogYearsld){
+    dogYear *= 7;
+    console.log("Your dog will be " + dogYear + " dog years old in " + dogYearFuture);
+} else{
+    console.log("Your dog will be " + dogYear + " human years old in " + dogYearFuture);
+}
+
+dogYearOfBirth = 2020;
+dogYearFuture = 2030;
+dogYear = dogYearFuture - dogYearOfBirth;
+shouldShowResultInDogYearsld = false;
+
+if(shouldShowResultInDogYearsld){
+    dogYear *= 7;
+    console.log("Your dog will be " + dogYear + " dog years old in " + dogYearFuture);
+} else{
+    console.log("Your dog will be " + dogYear + " human years old in " + dogYearFuture);
+}
+
+/*Housey Pricey*/
+
+
+var houseDepth = 10;
+var houseWidth = 8;
+var houseHeight = 10;
+var gardenSizeInM2 = 100;
+var volumeInMeters = houseDepth * houseWidth * houseHeight;
+var housePrice = volumeInMeters * 2.5 * 1000 + gardenSizeInM2 * 300;
+var friendPayment = 2500000;
+
+if(housePrice - friendPayment < 0){
+    console.log("He/she is overpaying, the price should be " + housePrice);
+}else{
+    console.log("He/she is underpaying, the price should be " + housePrice);
+}
+
+
+houseDepth = 11;
+houseWidth = 5;
+houseHeight = 8;
+gardenSizeInM2 = 70;
+volumeInMeters = houseDepth * houseWidth * houseHeight;
+housePrice = volumeInMeters * 2.5 * 1000 + gardenSizeInM2 * 300;
+friendPayment = 1000000;
+
+if(housePrice - friendPayment < 0){
+    console.log("He/she is overpaying, the price should be " + housePrice);
+}else{
+    console.log("He/she is underpaying, the price should be " + housePrice);
+}
+
+/*Ex Namey*/
+var firstWords = ["Forward", "Legend", "Good", "Waking", "Starting", "Vision", "Great", "Success", "Conquer", "Happy" ];
+var secondWords = ["Company", "Place", "Stay", "Oasis", "Career", "Ahead", "Future", "Colleague", "Dream", "Corporation"];
+var startupName = firstWords[Math.floor(Math.random() * 10)] + " " + secondWords[Math.floor(Math.random() * 10)];
```

Comment:
it's good practice to avoid repetitions in code. For example:
```suggestion
index=Math.random() * 10;
const startupName = firstWords[index] + " " + secondWords[index];
```
What do you think ?

Reply:
I get what you mean, but if I do it like that, then it will have the same index for both arrays, and then the random combination will not go around with all the possible options.

Reply:
Ah. Yes. you are right. Maybe as an improvement try not to use constants when you are accesing arrays
would be better to have something like  `Math.floor(Math.random() *firstWords.length) `

Reply:
Ok, I'll do that.